pkg/resources/kafka/configs.go

```diff
@@ -0,0 +1,205 @@
+		if err != nil {
+			return errors.WrapIfWithDetails(err, "could not alter broker config", "brokerId", brokerId)
+		}
+
+		// query the updated config
+		response, err := kClient.DescribePerBrokerConfig(brokerId, brokerConfigKeys)
+		if err != nil {
+			return errors.WrapIfWithDetails(err, "could not describe broker config", "brokerId", brokerId)
+		}
+
+		// update per-broker config status based on the response
+		if shouldUpdatePerBrokerConfig(response, fullPerBrokerConfig) {
+			return errorfactory.New(errorfactory.PerBrokerConfigNotReady{}, errors.New("configuration is out of sync"), "per-broker configuration updated")
+		} else {
+			statusErr := k8sutil.UpdateBrokerStatus(r.Client, []string{strconv.Itoa(int(brokerId))}, r.KafkaCluster, v1beta1.PerBrokerConfigInSync, log)
+			if statusErr != nil {
+				return errors.WrapIfWithDetails(err, "updating status for per-broker configuration status failed", "brokerId", brokerId)
+			}
+		}
+	} else if currentPerBrokerConfigState != v1beta1.PerBrokerConfigInSync {
+		log.V(1).Info("setting per broker config status to in sync")
+		statusErr := k8sutil.UpdateBrokerStatus(r.Client, []string{strconv.Itoa(int(brokerId))}, r.KafkaCluster, v1beta1.PerBrokerConfigInSync, log)
+		if statusErr != nil {
+			return errors.WrapIfWithDetails(err, "updating status for per-broker configuration status failed", "brokerId", brokerId)
+		}
+	}
+
+	return nil
+}
+
+func (r *Reconciler) reconcileClusterWideDynamicConfig(log logr.Logger) error {
+	kClient, err := kafkaclient.NewFromCluster(r.Client, r.KafkaCluster)
+	if err != nil {
+		return errorfactory.New(errorfactory.BrokersUnreachable{}, err, "could not connect to kafka brokers")
+	}
+	defer func() {
+		if err := kClient.Close(); err != nil {
+			log.Error(err, "could not close client")
+		}
+	}()
+
+	configIdentical := true
+
+	currentConfig, err := kClient.DescribeClusterWideConfig()
+	if err != nil {
+		return errors.WrapIf(err, "could not describe cluster wide broker config")
+	}
+	parsedClusterWideConfig := util.ParsePropertiesFormat(r.KafkaCluster.Spec.ClusterWideConfig)
+	if len(currentConfig) != len(parsedClusterWideConfig) {
+		configIdentical = false
+	}
+	if configIdentical {
+		for _, conf := range currentConfig {
+			if val, ok := parsedClusterWideConfig[conf.Name]; ok {
+				if val != conf.Value {
+					configIdentical = false
+					break
+				}
+			}
+		}
+	}
+
+	if !configIdentical {
+		err = kClient.AlterClusterWideConfig(util.ConvertMapStringToMapStringPointer(parsedClusterWideConfig), true)
+		if err != nil {
+			return errors.WrapIf(err, "validation of cluster wide config update failed")
+		}
+
+		err = kClient.AlterClusterWideConfig(util.ConvertMapStringToMapStringPointer(parsedClusterWideConfig), false)
+		if err != nil {
+			return errors.WrapIf(err, "could not alter cluster wide broker config")
+		}
+	}
+
+	return nil
+}
+
+func getBrokerConfigsFromConfigMap(configMap *corev1.ConfigMap) map[string]string {
```

Comment:
This function seems to perform the same thing as `util.ParsePropertiesFormat`.

Reply:
Indeed, did not see this function. Remove the function and the corresponding tests.